File: src/test/java/com/raychenon/leetcode/ValidParenthesesTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-09-04
 * https://leetcode.com/problems/valid-parentheses
 */
public class ValidParenthesesTest {

    private static ValidParentheses validParentheses;

    @BeforeClass
    public static void setUp() {
        validParentheses = new ValidParentheses();
    }

    @Test
    public void validAllParenthesesTest() {
        assertEquals(true, validParentheses.isValid("()"));
        assertEquals(true, validParentheses.isValid("[]"));
        assertEquals(true, validParentheses.isValid("{}"));
        assertEquals(true, validParentheses.isValid("<>"));
        // with space
        assertEquals(true, validParentheses.isValid(" ( )"));
    }

    @Test
    public void startByClosingTest() {
        assertEquals(false, validParentheses.isValid("]["));
    }

    @Test
    public void coupleOfParentheses() {
        assertEquals(true, validParentheses.isValid("[]()"));
    }

    @Test
    public void emptyStringTest() {
        assertEquals(true, validParentheses.isValid(null));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/ValidParenthesesTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * User: raychenon
 * Date: 2019-09-05
 * https://leetcode.com/problems/valid-parentheses/
 */
class ValidParenthesesTest {

    @Test
    fun validAllParenthesesTest() {
        assertEquals(true, ValidParentheses.isValid("()"))
        assertEquals(true, ValidParentheses.isValid("[]"))
        assertEquals(true, ValidParentheses.isValid("{}"))
        assertEquals(true, ValidParentheses.isValid("<>"))
        // with space
        assertEquals(true, ValidParentheses.isValid(" ( )"))
    }


    @Test
    fun validNonOnlyParenthesesTest() {
        // with space
        assertEquals(true, ValidParentheses.isValid(" ( )"))
        // with character
        assertEquals(true, ValidParentheses.isValid("[a]"))
    }


    @Test
    fun startByClosingTest() {
        assertEquals(false, ValidParentheses.isValid("]["))
    }

    @Test
    fun coupleOfParentheses() {
        assertEquals(true, ValidParentheses.isValid("[]()"))
    }

    @Test
    fun emptyStringTest() {
        assertEquals(true, ValidParentheses.isValid(""))
    }
}